addons/mass_mailing/static/src/js/mass_mailing_html_field.js

```diff
@@ -0,0 +1,574 @@
+            // In full width form mode, ensure the snippets menu's scrollable is
+            // in the form view, not in the iframe.
+            this.wysiwyg.snippetsMenu.$scrollable = this.wysiwyg.$el.closestScrollable();
+            // Ensure said scrollable keeps its scrollbar at all times to
+            // prevent the scrollbar from appearing at awkward moments (ie: when
+            // previewing an option)
+            this.wysiwyg.snippetsMenu.$scrollable.css('overflow-y', 'scroll');
+        }
+
+        // Remove the web editor menu to avoid flicker (we add it back at the
+        // end of the method)
+        this.wysiwyg.$iframeBody.find('.iframe-utils-zone').addClass('d-none');
+
+        // Filter the fetched templates based on the current model
+        const args = this.props.filterTemplates
+            ? [[['mailing_model_id', '=', this.props.record.data.mailing_model_id[0]]]]
+            : [];
+
+        const rpcQuery = buildQuery({
+            model: 'mailing.mailing',
+            method: 'action_fetch_favorites',
+            args: args,
+        })
+        // Templates taken from old mailings
+        const result = await this.rpc(rpcQuery.route, rpcQuery.params);
+        const templatesParams = result.map(values => {
+            return {
+                id: values.id,
+                modelId: values.mailing_model_id[0],
+                modelName: values.mailing_model_id[1],
+                name: `template_${values.id}`,
+                nowrap: true,
+                subject: values.subject,
+                template: values.body_arch,
+                userId: values.user_id[0],
+                userName: values.user_id[1],
+            };
+        });
+
+        const $snippetsSideBar = this.wysiwyg.snippetsMenu.$el;
+        const $themes = $snippetsSideBar.find("#email_designer_themes").children();
+        const $snippets = $snippetsSideBar.find(".oe_snippet");
+        const selectorToKeep = '.o_we_external_history_buttons, .email_designer_top_actions';
+        // Overide `d-flex` class which style is `!important`
+        $snippetsSideBar.find(`.o_we_website_top_actions > *:not(${selectorToKeep})`).attr('style', 'display: none!important');
+
+        if (device.isMobile) {
+            $snippetsSideBar.hide();
+            this.$content.attr('style', 'padding-left: 0px !important');
+        }
+
+        if (!odoo.debug) {
+            $snippetsSideBar.find('.o_codeview_btn').hide();
+        }
+        const $codeview = this.wysiwyg.$iframe.contents().find('textarea.o_codeview');
+        $snippetsSideBar.on('click', '.o_codeview_btn', () => {
+            this.wysiwyg.odooEditor.observerUnactive();
+            $codeview.toggleClass('d-none');
+            this.wysiwyg.getEditable().toggleClass('d-none');
+            this.wysiwyg.odooEditor.observerActive();
+
+            if ($codeview.hasClass('d-none')) {
+                this.wysiwyg.setValue($codeview.val());
+            } else {
```

Comment:
> What does that do / where does the val come from? From what I understand wysiwyg.setValue is essentially a .html?

The val come from the textarea codeview. Yes `setValue` will call `innerHTML`.

The codeview is the textarea that allows to change the html by hand.
![](https://i.imgur.com/Yc8MO31.png)

I can't reproduce the freezing problem. However I could reproduce a on the "plain text" template.
As this PR seems urgent, we open another task for that bug.

Reply:
@xmo-odoo

Reply:
> I can't reproduce the freezing problem. However I could reproduce a bug on the "plain text" template.

The lag, and probably the eventual crash, reproduces systematically in Firefox and FDE.

Reply:
@xmo-odoo I cannot reproduce the lag nor the crash in Firefox either. I can see it happen in your video but it is not clear to me what steps are causing the issue as I do not observe the same behavior on my machine. Since you can reproduce it systematically, could you please elaborate on what are the minimal steps that you have to take to reproduce it ? Thanks !

Reply:
1. open https://19126954-master-all.runbot84.odoo.com/
2. log in as admin
3. open "email marketing"
4. "create"
5. select "plain text"
6. put cursor in mail body
7. press `/`
8. confirm

everything starts lagging, memory usage starts creeping up, after some time an OS popup appears indicating firefox is not responding.

I tried capturing a profile, sadly the browser is in such a state that it does not work. The devtools UI seems to indicate that an error occurs, but the console never gets to actually displaying the error.

Reply:
I am able to reproduce the bug with these steps. Thank you !